src/app/core/components/home-page/home-page.component.ts

```diff
@@ -13,13 +15,32 @@ import {ErrorService} from '../../../shared';
 })
 export class HomePageComponent implements OnInit {
 
+  userName = '';
+
   constructor(private dialog: MatDialog,
               private authService: AuthService,
+              private userService: UserService,
               private errorService: ErrorService
   ) {
   }
 
   ngOnInit() {
+
+    this.userService.getUserInfoByToken(this.authService.getAccessToken())
+      .pipe(
+        catchError((err) =>
+          this.errorService.handleError(err)
+        )
+      )
+      .subscribe(() => {
+        if (this.userService.userInfo.subscribe()) {
+          this.userService.userInfo.subscribe(name => this.userName = name.name);
+        } else {
+          this.userService.getUserInfoByToken(this.authService.getAccessToken()).subscribe();
+          this.userService.userInfo.subscribe(name => this.userName = name.name);
```

Comment:
better move this.authService.getAccessToken() to variable

Reply:
done